### PR TITLE
ROX-17956: Fix error in roxctl authorize mode

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Currently, when executing `roxctl central login` and the central instance doesn't have any auth provider configured besides the default basic auth provider, the authorize roxctl page will contain an error.

This is due to the current login page code assuming that there will be at least one auth provider within the list of available auth providers. However, since we filter out the basic auth provider in roxctl authorize mode, since its not compatible with the flow, this assumption doesn't hold true anymore.

This PR fixes this by a) ensuring that the current code in `LoadingOrForm` respects potential empty auth provider list within the initial values and b) returning an error in roxctl authorize mode in case of an empty auth provider list after filtering out the basic auth provider.

Screenshot 2023-06-20 at 18 36 33

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6584** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6584" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6584?utm_source=stack-comment).

File: ui/apps/platform/src/Containers/Login/LoginPage.js [hunk 1]
```diff
@@ -174,6 +174,19 @@ class LoginPage extends Component {
         if (authorizeRoxctlMode) {
             authProviders = authProviders.filter((provider) => provider.type !== 'basic');
             title = 'Authorize roxctl';
+            if (authProviders?.length === 0) {
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Login/LoginPage.js:177]: Question that is not a blocker to merge: is `?` optional chaining needed here?

It seems like `authProviders.filter(…)` call above would throw if the value is nullish.

File: ui/apps/platform/src/Containers/Login/LoginPage.js [hunk 2]
```diff
@@ -306,7 +319,7 @@ const LoadingOrForm = ({ authProviders, authorizeRoxctlMode = false }) => {
     }
 
     const options = authProvidersToSelectOptions(availableAuthProviders);
-    const initialValues = { authProvider: options[0].value };
+    const initialValues = { authProvider: options[0]?.value };
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Login/LoginPage.js:322]: In contrast to the other comment, this makes sense.

How about a comment to explain when the array might be empty?

**Review by pedrottimark — APPROVED** [review]
2 comments for your consideration.

**Comment by jschnath** [comment]
@dhaus67 if this is ready on your end please merge it asap so I can cut what's looking like the final RC

**Pull request merged**